lib/cdo/delete_accounts_helper.rb

```diff
@@ -224,4 +224,16 @@ def purge_user(user)
     user.purged_at = Time.zone.now
     user.save(validate: false)
   end
+
+  # Given an email address, locates all accounts (including soft-deleted accounts)
+  # associated with that email address and purges each of them in turn.
+  # @param [String] email an email address.
+  def purge_all_accounts_with_email(email)
+    # Note: Not yet taking into account parent_email or users with multiple
+    # email addresses tied to their account - we'll have to do that later.
+    (
+      User.with_deleted.where(email: email) +
+      User.with_deleted.where(hashed_email: User.hash_email(email))
```

Comment:
These *might* eventually need rewriting in light of multi-auth, but I'm not sure yet - I started writing a `User.where` replacement that routes these sorts of lookups correctly, but getting it fully right was a bit trickier than I'd initially anticipated.